src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -467,7 +469,8 @@ internal Enumerator(ReadOnlyTensorSpan<T> span)
                 _span = span;
                 _items = -1;
                 _curIndexes = new nint[_span.Rank];
-                _curIndexes[_span.Rank - 1] = -1;
+                if (_span.Rank > 0)
```

Comment:
What is this fixing? @michaelgsharp, is there a bug here that needs to be fixed in 9?

Reply:
Yes, this fixes a bug in creating an enumerator for an empty (default) Tensor. I've added tests for this case.